pkg/zkproofs/range.go

```diff
@@ -0,0 +1,144 @@
+package zkproofs
+
+import (
+	crand "crypto/rand"
+	"encoding/json"
+	"errors"
+	"github.com/coinbase/kryptology/pkg/bulletproof"
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	"github.com/gtank/merlin"
+	"github.com/sei-protocol/sei-cryptography/pkg/encryption/elgamal"
+)
+
+const rangeDomain = "rangeDomain"
+const ippDomain = "ippDomain"
+
+type RangeProof struct {
+	Proof      *bulletproof.RangeProof
+	Randomness curves.Point
+	UpperBound int
```

Comment:
I added this initially, but on second thought it might make sense for the verifier to pass in the UpperBound themselves.

The verifier would say: I want to verify that this ciphertext encrypts some value > 0 and below 2^upperBound
This is codified as VerifyRange(ciphertext, upperBound, Proof)

If this is included in the Proof object, verifier needs to make sure that the upper bound provided by the proof matches the upper bound they are trying to verify for.

Reply:
Or actually maybe we leave it in the RangeProof object to let the verifier know what range this proof can be used for.